### DISCUSSION TITLE
Raspberry Pi - Not Synced

### DISCUSSION THREAD

**Discussion by OptimAdam** [Chia-Network/chia-blockchain#4769] [Support]
Hi,

Is anyone running the full node on a Raspberry pi 4?

It was running ok for me for a while, but recently its really struggling to keep up with the block chain and it is always falling out of sync. It would be good to know if anyone else is experiencing this or still using a pi.

I am thinking the chain is now to big for the Pi to process Quite often I see the Full node process running at 100% CPU.

Pi is running Ubuntu 20.04.2 LTS

**Comment by jduraes** [comment]
Yep, same happening here. I've only noticed it today, but could have been happening for a few days, I fear.

And I am running mine on Raspberry Pi OS, on a Pi 4 with 8GB RAM which I bought specifically for this... :-(

![image](https://user-images.githubusercontent.com/5991582/118375347-0dcc4480-b5b9-11eb-9d6a-e7c2c53be2a3.png)

Full Node, firewall ports open and correctly forwarded, ~50 nodes connected and up-to-date.

Has to be the efficiency of the daemon taxing the Pi, I also believe:

![image](https://user-images.githubusercontent.com/5991582/118375395-48ce7800-b5b9-11eb-8d77-4dd72d787fed.png)

**Reply by OptimAdam** [reply]
I have just upgraded my SD Card to see if that has any difference.

**Reply by jduraes** [reply]
Cool. Please let us know as I have one on order as well.

**Reply by OptimAdam** [reply]
I bought the SanDisk Extreme Pro. The Pi does seem to be running better now, but still experiencing issues.

I am going to rebuild the OS and see if it makes a difference.

**Reply by jduraes** [reply]
Ha, I got the same! :-)

**Reply by forrestcamps** [reply]
So guys, Any news here? I have one raspberry pi 4 too, 4gb. 32gb SD with raspbian, I'm think in future let him farming for me, with low energy spending. I'm using my macbook pro to plotting. So I wanna know how to start a workflow with doing the plots here on mac, and send by network to my raspberry pi and leave the plots farming there. There is a way for a noob in Linux, Cli at all. (I,m running with a GUI on mac. that's how I can do for a while. Thank you. best wishes from brazil.

**Reply by jduraes** [reply]
Nao forrestcamps - so' recebo o meu uSD amanha, portanto nao tenho novidades. 

Back to English now. If you are thinking about transferring the plots to the Pi over a network connection, think again. That will take forever. The Pi 4 ethernet port runs at 943Mbps max (https://cdn.mos.cms.futurecdn.net/g4HHurrhBE7CGzBYN9x7kM.png) - and on top of that, it uses the built in USB-3 hub, which runs at ~6Gbps. So these max speeds will be shared, so transferring ~100GB per plot is not a trivial thing. And if you happen to start transferring a second plot before the first one finishes, well then, I'm not sure how the Pi is going to be able to cope with that much I/O.

I have my plots stored on a NAS, so the files don't need to be transferred to the Pi itself. But I'm starting to wonder if this is a wise strategy, as I don't yet fully understand how the algorithm to check the Proofs works. And could it be this that causes the Pi to struggle in the first place?

Finally, as I was reading more info just now, I've realised that the Pi uSD slot is only capable of ~50Mbps, so having this SanDisk Extreme Pro (170Mbps read) may not actually help much. I think I might cut my losses and move to an SSD over USB-3 instead for the OS.

I'll think about it...

**Comment by nunoluis** [comment]
I have exactly the same problem. I described it here https://github.com/Chia-Network/chia-blockchain/discussions/5223

I had my node running well on a desktop, and just moved everything to a Pi (including the synced blockchain db) and cannot be in sync anymore.

Plus getting a very strange readout from the debug.log file....

please help!

**Comment by juls16** [comment]
Same setup, although running with Raspbian OS 64 and booting from an external SSD. It just cannot keep up and loses connection.

**Comment by OptimAdam** [comment]
That`s disappointing to hear.

So I am having the same issue on another system I built.

Running on my Synology NAS I am running Windows Server Core with 3GB ram (40% usage), 3 core (6 threads) 1.5GHz Celeron, CPU normally running around 75% usage and It can`t keep up.

I really want to know what specs of hardware you need to run this on. I feel the single core speed makes the difference.

I am running another test to see if my Windows 10 Desktop VM with 4GB and 3 core (6 Threads) 3.6Ghz can keep up

**Reply by OptimAdam** [reply]
That`s not the issue.

I have run them individual for 8 hours over night and it makes no difference. The system with more cores and fast core speed always syncs and catches up, the system with less cores and slower core speed is always behind.

I have also run 2 on the same network with the same 12 core CPU and they ran fine.

**Reply by climba03003** [reply]
I am running docker inside my Synology NAS, and it always synced. I think core speed do not affect but good peer connection does.

Basically, it consume around 10% CPU and 1GB of ram.
NAS model: DS1621+

**Reply by OptimAdam** [reply]
Which Synology NAS are you using?

**Reply by climba03003** [reply]
DS1621+

**Reply by OptimAdam** [reply]
That model has a CPU 2.2GHz (Turbo Boost up to 2.7GHz). 

So it can`t be ruled out.

**Reply by climba03003** [reply]
10% Usage on 2.2GHz vs 75% Usage on 1.5GHZ
That's a great diff. and Docker only focus on Chia but Windows Server Core have other staff to do.

**Reply by OptimAdam** [reply]
The difference is if you cant burst up to finish a task and there is a long list of task, they all wait and execute after each other. So the pipeline is always going to be maxed out.

If the CPU can finish one task quicker and then waits you will see a dramatic drop in load. Difference between single threaded and multi threaded performance. Its an exponential increase in load and it will always be trying to catch up.

I current see the same issue on the Pi running Ubuntu, so its not just on windows.

Can you down clock or limit the CPU cycles of the container?

**Reply by climba03003** [reply]
I have another computer running "Windows Server" with Xeon Silver 4108
4% CPU consumption with 1.8GHz
That is the closest I have.

**Reply by OptimAdam** [reply]
I have shut every other system down on my network and port forwarding 8444 to the system. Lets see if it can catch up. 

I would love to know how you got the docker container working on the Synology, I ran into a few issues with it.

FYI, all server core processes add upto 5% CPU

**Reply by climba03003** [reply]
1. SSH inside NAS
2. Run `docker pull ghcr.io/chia-network/chia:latest`
3. Go to the web dashboard
4. Launch with the image

Important Mount Volumne
`ro` means Read-Only and `rw` means Read-Write
`ro - /plots` Plots directory for farming
`ro - /<path to key file>` Consistence the key
`rw - /root/.local/share/python_keyring ` Prevent resync after shutdown of docker
`rw - /root/.chia` Prevent resync after shutdown of docker

**Reply by OptimAdam** [reply]
So it has been running for 6 hours and its now further behind the chain than it was when it started. Started today around 300500 now at 302978. Looks like I need to get to 306467.

The system has connected to 30 nodes, downloading of one node at 22  MiB. I see the full_node process just spiking a lot of the time and running at 45 - 75% of the CPU. Memory has been sitting at 60% most of the day, my bandwidth is 300 and is hardly doing anything 

![image](https://user-images.githubusercontent.com/17952289/118889704-584c0a80-b8b2-11eb-891c-570f8a3c2163.png)

Running out of ideas

**Reply by nunoluis** [reply]
Yes, this is what I've seen also with my Pi4: short bursts trying to catch up, with CPU usage around 100% on all 4 cores, but overall falling more and more behind the chain and never achieving to be fully in sync. 
I think we need a more detailed analysis of this, of people that have the same setup (Pi4 and not a RockPi - which as I understand is more powerful) to tell what their settings are in config.yaml (max peers, wallet sync or not? and other settings).

I tried keybase, but couldn't get anything that worked from there (usual tips, open 8444 port, reduce max peers).

Very frustrating that I got a Pi4 just for this. It worked fine before transaction , but now nope...

**Comment by juls16** [comment]
Update:
I noticed that my CPU usage was always at 100% so I decreased the total number of peers it tries to connect to. It seems that it has some effect for some poeple with the rpi but not in my case, unfortunately. 

EDIT: only one core is at 100%

**Reply by OptimAdam** [reply]
That's why I think its based on the individual core speed, rather than core count.

**Reply by juls16** [reply]
Yep

**Comment by jduraes** [comment]
Disable uPNP - it is not a requirement. It will sync fine without it, with
fewer peers. I have 4 machines on the same network, all synching fine. Only
one has port-forwarding configured.

**Reply by juls16** [reply]
Done that but it did not change anything.

**Reply by jduraes** [reply]
Oh no, sorry, I wasn't saying it would fix the issue regarding the Pi/low core speed not keeping up with it. I was answering the uPNP comment that someone else made - which is no longer in this thread...

**Reply by juls16** [reply]
ah okay thanks. Anyway, I did it before your post so no worries..

**Comment by Vince-nl** [comment]
I have the same sync problems with my raspberry pi 4 (4gb) as you guys. Has anyone tried the new version of Chia (1.1.6) and a Rasperberry Pi yet? Sync should be better with this version.

**Reply by nunoluis** [reply]
I updated this morning and tried to re-sync my Pi4. I was around 291000 (313000 for the chain) and the Pi4 still showed the same issue: very slow moving along the blocks, high CPU usage and acting in bursts (often not advancing any block, then a lot at the same time, and repeat).

I reduced max_peers to 10, and was connected to nodes all with >313000, so that should not be issue. I noticed with 1.1.6 the peers I connected to were all high quality (i.e. synced or almost synced to the chain).

**Reply by Vince-nl** [reply]
Ah nice okay... so, max_peers to 10 could be a solution with the new 1.1.6 chia version? Or is it too early to say that?

**Reply by juls16** [reply]
It seems for me that 1.1.6 fixed everything. It synced very quickly, along with my wallet and it does not even miss any challenges !! Note that I decreased target_peer_count to 3, previously. I will try to increase it a bit, to see if it is working.

**Reply by Vince-nl** [reply]
Nice, do you guys running the GUI of Chia or just in terminal?

**Reply by nunoluis** [reply]
Mmm... max 3 peers...? I'll give it a try. I left it at 10 when I left my house in the morning, but seeing how slowly it was syncing I don't have many hopes it will be farming when I get back later. I will try to reduce it further from 10 to 3 max peers and see if that helps.

I'm running the GUI at all times for farming (use the CLI only to launch plotting jobs on another machine). The Pi4 is running Raspbian 64 (the latest beta release). I also thought about installing a headless Ubuntu 20.04 to see if that would help...?

@juls16 which OS are you running? Can you provide some more specs on your setup? Also, when you say you synced quickly, from which block was that? From scratch, a fresh install? From 20000 blocks behind the latest block? I'm almost considering erasing everything and re-install, but afraid that the ~400000 block syncing process will take forever...

**Reply by juls16** [reply]
I am running Rasbpian 64 on RPi 4 8gb. Note that it is not booting from the SD card but from an external SSD that I had laying around (Samsung T5 500GB). I run chia using the CLI, only.

It synced the blockchain pretty quickly from 15000 block behind and the wallet from around 100000 blocks behind. I don't know exactly how long it took. But after updating from 1.1.5 (so no fresh install), I started the farmer and noticed in the debug that evrything was syncing really fast compared to what I experienced before. I left and came back 4 hours later and everything was synced, farming smoothly.

400000 will take some time. Try to run the full node on a different computer. Once this is done transfer the blockchain and wallet db's into your pi.

**Reply by nunoluis** [reply]
thanks for the info. I got home and my Pi4 is synced, so it does seem that 1.1.6 helps with that. In between blocks I still wait longer than with my gaming desktop (2-3 min without anything showing up in the GUI and then 3 or 4 block entries almost at once) but I'm getting challenges roughly 10-12 seconds apart, so that looks fine.

A little off topic, does anyone know what the "not farmed by us" in the log file means? (log set to INFO, and I'm greping for "time"):

![Screenshot 2021-05-21 at 17 57 17](https://user-images.githubusercontent.com/56304966/119166469-f4b31000-ba5e-11eb-8c01-34187c3c529d.jpg)

**Comment by OptimAdam** [comment]
I don't know what fixed it but I did a complete reinstall with 1.1.6. 

I reimaged the sdcard and reinstall ubuntu and set everything up again. This included download a new version of the block chain instead of coping it from an different source.

**Reply by Vince-nl** [reply]
I did also a complete reinstall of Ubuntu and Chia 1.1.6 on my Raspberry Pi and I have a full node after syncing days. So that's good news. 

But when I use the command "chia wallet show" it says my wallet is out of sync.

Do you guys have that too?

**Reply by OptimAdam** [reply]
It is staying in sync for me

![image](https://user-images.githubusercontent.com/17952289/119688250-abffac00-bdfc-11eb-92a7-1f4bf4eea958.png)

Wallet also says Status: Synced
![image](https://user-images.githubusercontent.com/17952289/119688492-e10bfe80-bdfc-11eb-99dc-c71b06c8704c.png)

**Reply by Vince-nl** [reply]
Did you do something different? Opening a another port (i have port 8444 open)? I don’t have connected a harddisk with plots because I first wanted to know if I got a full node. Can that be the reason of a not synced wallet?

**Reply by nunoluis** [reply]
Hi @Vince-nl , I will check if my wallet is synced once I get home, but I can  confirm that my Pi4 running Raspbian and 1.1.6 was able to sync throughout one day and afterwards I even minted 1 block after that, so things look fine.

As a potential issue, could you be running the Pi4 node from an IP where you have another node connected? I ask because you write "I don’t have connected a harddisk with plots because I first wanted to know if I got a full node" - if you have your HDDs connected to another computer/farmer/node using the same IP this can create problems (actually, it should not even work, as the rule is 1 node per IP).

**Reply by Vince-nl** [reply]
Hi nunoluis,

Thnx for your reply!

_"I will check if my wallet is synced once I get home, but I can confirm that my Pi4 running Raspbian and 1.1.6 was able to sync throughout one day and afterwards I even minted 1 block after that, so things look fine."_

Cool thank you, I'm curious.

I had a game pc that I farmed with as well as a plotter. But I bought a Raspberry Pi for farming so I can use my game computer for plotting only.
I’ve installed Chia on my game pc but I don’t have it open in Windows. I’m plotting with Swar's Chia Plot Manager on my game computer so I don’t run the Chia blockchain, right?

Or can you still have complications of you do that?

**Reply by nunoluis** [reply]
All good, wallet is in sync, just like the blockchain.

I did as you @Vince-nl, I have a better PC that I used for plotting and farming, and when I reached my max HDD space (not a lot...) I moved the drives to a Pi4. I had done this before transactions went live and it all worked well, but the netspace was smaller and not a lot was happening on the chain.

When I did it again last week, as I explained above, things were very very slow (version 1.1.5). Once 1.1.6 came out, I reduced max_peers in the config.yaml file to 10 and once the Pi4 got in synced all was well and stayed that way.

Hope you figure it out. Good luck!

**Reply by Vince-nl** [reply]
I have set max_peers 3, maybe I have to increase it to 10. I will try.

**Reply by Vince-nl** [reply]
I don't know what happend but I have a synchronised wallet now :-)
I did nothing but I think that patience is/was the solution :)

Thank you all for the help and suggests.

**Reply by nunoluis** [reply]
Great! Good luck! :)

**Comment by BrandtH22** [comment]
@OptimAdam , based on your previous message it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
